e/php/bootstrap

```diff
@@ -16,42 +16,61 @@ $lambdaRuntimeApi = getenv('AWS_LAMBDA_RUNTIME_API');
 require __DIR__ . '/vendor/autoload.php';
 
 $socket = '/tmp/php-fpm.sock';
-$config =  __DIR__ .'/php-fpm.conf';
-$cgi = new Process("php-fpm --fpm-config {$config} -c /opt/php.ini -d extension_dir=/opt/php/extensions");
-$cgi->setTimeout(null);
-$cgi->start(function ($type, $output) {
-    if ($type === Process::ERR) {
-        echo($output);
-        exit(1);
-    }
-});
-
-
-register_shutdown_function(function() use(&$cgi) {
-    $cgi->stop();
-});
-
-$wait = 5000; // 5ms
-$timeout = 5000000; // 5 secs
-$elapsed = 0;
-while (! file_exists($socket)) {
-    usleep($wait);
-    clearstatcache();
-    $elapsed += $wait;
-    if ($elapsed > $timeout) {
-        echo "Timeout while waiting for socket at {$socket}..";
-        exit(1);
+
+function is_server_started($socket)
+{
+    return file_exists($socket);
+}
+
+function wait_for_server($socket)
+{
+    $wait = 5000; // 5ms
+    $timeout = 5000000; // 5 secs
+    $elapsed = 0;
+    while (! is_server_started($socket)) {
+        usleep($wait);
+        clearstatcache($socket);
+        $elapsed += $wait;
+        if ($elapsed > $timeout) {
+            echo "Timeout while waiting for socket at {$socket}..";
+            exit(1);
+        }
     }
+
+    echo "FastCGI started on {$socket}";
+}
+
+function start_server()
+{
+    $config =  __DIR__ .'/php-fpm.conf';
+    $cgi = new Process("php-fpm --fpm-config {$config} -c /opt/php.ini -d extension_dir=/opt/php/extensions --nodaemonize");
+    $cgi->setTimeout(null);
+    $cgi->start(function ($type, $output) {
+        if ($type === Process::ERR) {
+            echo($output);
+            exit(1);
+        }
+    });
+
+    register_shutdown_function(function() use(&$cgi) {
+        $cgi->stop();
+    });
 }
 
-echo "Fastcgi socket running on {$socket}";
+start_server();
+wait_for_server($socket);
 
-$client  = new Client(new UnixDomainSocket($socket));
+$client  = new Client(new UnixDomainSocket($socket, 30000));
```

Comment:
@barryvdh UnixDomainSocket receives 3 parameters, the 2nd one is the Connection Timeout, This i think we can let as the default 5s.
The 3rd one is the R/W Timeout, this one should be 30s(or Xs, i am saying 30 because this is PHPs Default)

`$client  = new Client(new UnixDomainSocket($socket, 5000,30000));
`

Reply:
Fixed, thanks